packages/app/src/components/nominationInfo/ApplicationForm.js

```diff
@@ -0,0 +1,311 @@
+import { useForm, Controller } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as Yup from 'yup';
+import nominationsAPI from '../../utils/API/nominationsAPI';
+import { NominationsDataContext } from '../../utils/context/NominationsContext';
+import { ActiveNominationContext } from '../../utils/context/ActiveNominationContext';
+import DatePicker, { CalendarContainer } from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+const ApplicationForm = props => {
+  const firstUpdate = useRef(true);
+
+  const [NominationsData, setNominationsData] = useContext(
+    NominationsDataContext
+  );
+
+  const [activeNomination, setActiveNomination] = useContext(
+    ActiveNominationContext
+  );
+
+  const [admissionDate, setAdmissionDate] = useState(new Date());
+  const [dischargeDate, setDischargeDate] = useState(new Date());
+
+  useEffect(() => {
+    if (props.formData['Admission Date'] != 'Invalid Date') {
+      setAdmissionDate(props.formData['Admission Date']);
+    }
+  }, [props.formData['Admission Date']]);
+
+  useEffect(() => {
+    if (props.formData['Discharge Date'] != 'Invalid Date') {
+      setDischargeDate(props.formData['Discharge Date']);
+    }
+  }, [props.formData['Discharge Date']]);
+
+  const phoneRegex = /^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$/;
+  const yesNoRegex = /^(?:Yes\b|No\b)/;
+
+  useEffect(() => {
+    if (!firstUpdate.current) {
+      handleSubmit(submitForm)();
+    }
+    firstUpdate.current = false;
+  }, [props.saveHasBeenClicked]);
+
+  const validationSchema = Yup.object({
+    'Admission Date': Yup.date().required('Required'),
+    'Discharge Date': Yup.date()
+      .min(
+        Yup.ref('Admission Date'),
+        'Discharge date cannot be before admission date.'
+      )
+      .required('Required'),
+    'Representative Name': Yup.string()
+      .min(3, 'Must be 3 characters or more.')
+      .max(30, 'Must be 30 characters or less.')
+      .required('Required'),
+    'Representative Email Address': Yup.string()
+      .email('Invalid email address.')
+      .required('Required'), // This handles email validation with no regex.
+    'Representative Phone Number': Yup.string()
+      .matches(phoneRegex, 'Please enter a valid phone number.')
+      .required('Required'),
+    Relationship: Yup.string()
+      .min(3, 'Must be at least 3 characters.')
+      .max(20, 'Must be no more than 20 characters.')
+      .required('Required'),
+    'Request to communicate in Spanish?': Yup.string()
+      .matches(yesNoRegex, 'Please enter yes or no.')
+      .required('Required'),
+  });
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(validationSchema),
+  });
+
+  const submitForm = async data => {
+    if (NominationsData) {
+      let newActiveNomination = {};
+      const newNominationData = NominationsData.map(nomination => {
```

Comment:
Could this use the active nomination from `ActiveNominationsContext` instead of looping through here?

Reply:
this issue is going to be revisited in #272 but if the nominationsData is not updated, then the command center nominations will not be updated. activeNominationsContext is a single nomination from nominationsData to my understanding. These are two different use context things. The activeNominationsContext is also updated at the end of submitForm function with the useContext function setActiveNomination.

Reply:
I don't understand the purpose of the ActiveNominationsContext really.

Reply:
we will design a better state management logic and implement it in #272

Reply:
What do you think of this or something like it @aschey https://github.com/supasate/connected-react-router#step-3 ?

Reply:
Or we can not change anything and only update active nomination in submit form then update nomination data on load of command center, that sounds like less work lol and more the way of how it is all set up now. We will revisit in #272.

Reply:
I think `activeNomination` is there just so you can find the nomination you want without filtering through all of them each time. There's probably room to improve the way this state is managed in general. I'd hesitate to introduce a dependency on Redux because we don't really have a ton of state to manage and in my experience Redux tends to add a lot of unnecessary complexity unless you have some complex state to manage. I don't know the code as well as you guys do but one relatively simple thing that might work is to use a custom hook that uses context internally. Something like this:
```js
const { setActiveNomination, activeNomination, setNominations, nominations } = useNominations();
setActiveNomination({...activeNomination, name: 'blah'})
```
This would update activeNomination and internally the hook would also find the correct entry in the array and update it so `activeNomination` and `nominations` are always in sync.

Reply:
we are doing this in the ticket https://app.zenhub.com/workspaces/ksf-5f23520d91f7ee00134cbaf6/issues/the-difference-engine/ksf/272 so can you remove this change requested? @aschey both contexts are being updated right now, but will be change later, do you really want to update the activeNomination only right now and not NominationData?

Reply:
Okay, we can leave this as is for now since it will be refactored in that other ticket.